.github/workflows/pull_requests.yml

```
@@ -0,0 +1,31 @@
+name: Pull Request
+
+on:
+  pull_request:
+    branches: [ main ]
+
+## We are cancelling previously triggered workflow runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+ 
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        swift:
+        - image: swift5.8.1-jammy
+        - image: swift5.9.2-jammy
+        - image: swift5.10.1-noble
```

Is this a private repository or did you mean:
```suggestion
        - image: swift:5.8-jammy
        - image: swift:5.9-jammy
        - image: swift:5.10-noble
```
This will automatically get the latest minor version of each image.

---

Ah I see you did mention this should use Apple's repository.